share/scripts/new_clean_playlists.lua

```diff
@@ -0,0 +1,260 @@
+ardour { ["type"] = "EditorAction", name = "New clean playlists",
+	license     = "MIT",
+	author      = "Mathieu Picot (Houston4444)",
+	description = [[Copy the current playlist of selected tracks to a new playlist with audible regions only.]]
+}
+
+function factory (params) return function ()
+
+	-- http://manual.ardour.org/lua-scripting/class_reference/#ArdourUI:Selection
+	local sel = Editor:get_selection ()
+
+	-- Track/Bus Selection -- iterate over all Editor-GUI selected tracks
+	-- http://manual.ardour.org/lua-scripting/class_reference/#ArdourUI:TrackSelection
+	for route in sel.tracks:routelist():iter() do
+
+		-- each of the items 'route' is-a
+		-- http://manual.ardour.org/lua-scripting/class_reference/#ARDOUR:Route
+		local track = route:to_track() -- see if it's a track
+		if track:isnil() then
+			-- if not, skip it
+			goto next_route
+		end
+        
+		local is_audio = false
+		if track:data_type():to_string() == "audio" then
+			is_audio = true
+		end
+
+		local main_pl_name = track:name() .. "._MAIN_"
+		local main_pl_ex_name = track:name() .. "_ExMAIN_"
+
+		-- find the playlist name track_name._MAIN_
+		for pl in Session:playlists():playlists_for_track (track):iter() do
+			if pl:name() == main_pl_name then
+				pl:set_name(main_pl_ex_name)
+				break
+			end
+		end
+
+		-- copy current playlist, get new playlist and rename it
+		-- http://manual.ardour.org/lua-scripting/class_reference/#ARDOUR:Playlist
+        track:use_copy_playlist()
+		local playlist = track:playlist()
+		playlist:set_name(track:name() .. "._MAIN_")
+
+		-- we will stock here the values needed to modify/remove regions
+		local regions_to_modify = {}
+
+		for r in playlist:region_list():iter() do
+			local r_front = r:position()
+			local r_end = r_front + r:length()
+
+			-- region can be finally splited into many segments
+			-- 'segments' is a table containing segments with the form
+			-- {{seg1_start, seg1_end}, {seg2_start, seg2_finish}...}
+			-- segments will finally be regions
+			local segments = {{r_front, r_end}}
+
+			-- values used for audio tracks to prevent cut during an audible fade
+			local r_no_cut_before = r_front
+			local r_no_cut_after = r_end
+
+			if is_audio then
+				local ra = r:to_audioregion()
+
+				if ra:fade_in_active() then
+					r_no_cut_before = r_front + ra:fade_in_length() + 64
+				end
+
+				if ra:fade_out_active() then
+					r_no_cut_after = r_end - ra:fade_out_length() - 64
+				end
```

Comment:
regions shorter than 128 samples are not copied to the new playlist, because an isolated region with a such length could only be noise. But at this point of the script we just check from when we could cut the compared regions.